RiotSwiftUI/Modules/Authentication/Common/Service/MatrixSDK/AuthenticationService.swift

```diff
@@ -309,7 +316,14 @@ class AuthenticationService: NSObject {
 }
 
 extension MXLoginSSOIdentityProvider {
-    var ssoIdentityProvider: SSOIdentityProvider {
-        SSOIdentityProvider(id: identifier, name: name, brand: brand, iconURL: icon)
+    @objc func makeSSOIdentityProvider(isOIDC: Bool) -> SSOIdentityProvider {
+        SSOIdentityProvider(id: identifier, name: name, brand: brand, iconURL: icon, isOIDC: isOIDC)
+    }
+}
+
+extension MXLoginSSOFlow {
+    @objc var ssoIdentityProviders: [SSOIdentityProvider] {
+        identityProviders.map { $0.makeSSOIdentityProvider(isOIDC: delegatedOIDCCompatibility)}
```

Comment:
I would question if this part is correct. AIUI, if there are multiple providers, then the presence of this flag means "don't show them anyway" but as we're not doing that, I would be surprised if these are then using OIDC compatibility (as they're basically Sign in with Google/Apple etc which aren't).